todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TasksMvpController.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import android.support.annotation.NonNull;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
+
+import com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailFragment;
+import com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailPresenter;
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksFragment;
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksPresenter;
+import com.example.android.architecture.blueprints.todoapp.util.ActivityUtils;
+
+/**
+ * Class that creates fragments (MVP views) and makes the necessary connections between them.
+ */
+public class TasksMvpController {
+
+    private final FragmentActivity mFragmentActivity;
+
+    private TasksTabletPresenter mTasksTabletPresenter;
+
+    private TasksPresenter mTasksPresenter;
+
+    private TaskDetailPresenter mTaskDetailPresenter;
+
+    /**
+     * Creates a controller for a task view for phones or tablets.
+     * @param fragmentActivity the context activity
+     * @return a TasksMvpController
+     */
+    public static TasksMvpController createTasksView(@NonNull FragmentActivity fragmentActivity) {
+        checkNotNull(fragmentActivity);
+        TasksMvpController tasksMvpController = new TasksMvpController(fragmentActivity);
+        tasksMvpController.initTasksView();
+        return tasksMvpController;
+    }
+
+    /**
+     * Creates a controller for a task detail view for phones or tablets.
+     * @param fragmentActivity the context activity
+     * @return a TasksMvpController
+     */
+    public static TasksMvpController createTaskDetailView(
+            @NonNull FragmentActivity fragmentActivity, String taskId) {
+        checkNotNull(fragmentActivity);
+        TasksMvpController tasksMvpController = new TasksMvpController(fragmentActivity);
+        tasksMvpController.initTaskDetailView(taskId);
+        return tasksMvpController;
+    }
+
+    // Force factory method, prevent direct instantiation:
+    private TasksMvpController(@NonNull FragmentActivity fragmentActivity) {
+        mFragmentActivity = fragmentActivity;
+    }
+
+    private void initTasksView() {
+        if (isTablet()) {
+            createTabletElements();
+        } else {
+            TasksFragment tasksFragment = findOrCreateTasksFragment();
+            if (tasksFragment.getPresenter() != null) {
+                mTasksPresenter = (TasksPresenter) tasksFragment.getPresenter();
```

Comment:
how about letting the `Controller` deal with this? the `Controller` could be attached to the `Activity` lifecycle and would be in charge of assigning the `Presenter` to each `Fragment`